Update error path for "Check transfer directory for objects"

jorikvankemenade opened the pull request:
This PR solves the two problems that are discussed in [#1108](https://github.com/archivematica/Issues/issues/1108):
- e51ca85 removes the "Check transfer directory for objects" as a terminal link. This should resolve the reported `RuntimeError` from showing up in the logs. 
- 2745bf8 configures links the "Check transfer directory for objects" to the failure path for the Transfer. This means that in case of failure an e-mail is sent to the user and the transfer is cleaned. Besides that, I have changed the error codes and code style. For a full explanation please check the commit message.

Using the Images transfer from the samples I have verified that I can Transfer/Ingest without the `RuntimeError` showing up in the logs.

![image](https://user-images.githubusercontent.com/7934339/75094749-5fd91280-558e-11ea-8310-1b36c7628c52.png)

When I run the same transfer, but make sure to clear the objects directory before hitting "Check transfer directory for objects", I get a failed transfer and a failure report:

![image](https://user-images.githubusercontent.com/7934339/75094830-fd344680-558e-11ea-8fd8-076ba6b79854.png)

![image](https://user-images.githubusercontent.com/7934339/75094838-0cb38f80-558f-11ea-8924-d62176a85ddd.png)

jorikvankemenade committed:
queues: update PACKAGE_COMPLETED_LINK_IDS

Remove "Check transfer directory for objects" as a terminal link. The
package is not yet completed when this link is hit. You either continue
to the "Create SIP from Transfer" microservice, or fail the transfer.

jorikvankemenade committed:
check_transer_dir_for_objects: set error path

When the transfer directory does not contain any objects we need to fail
the transfer. In case of a non zero exit code we need to fall back to
the "Email fail report" step. This step is also configured as the
default fallback in case anything else goes wrong.

This commit includes two small changes to the
check_transfer_dir_for_objects script. First of all it changes the exit
codes to more intuitive scheme. In case there are objects the exit code
is 0 (success) and the frontend will give a green exit code sign. In
case of no objects the script returns 1 and the frontend will display a
red exit code sign.

This commit also simplifies the if check for existing files. Files is a
is a list of the names of the non-directory files in the directory path.
This means that files is either an empty or non-empty list, and maybe
but not likely, None. In case the list is empty or None, there are no
files, otherwhise there are files. This behaviour is nicely captured by
the default boolean resolution of the Python list:

In [1]: files = []
In [2]: bool(files)
Out[2]: False

In [3]: files = ['foo', 'bar']
In [4]: bool(files)
Out[4]: True

In [5]: files = None
In [6]: bool(files)
Out[6]: False

This commit also transforms some variables in the script to snake_case
to match default Python code style and the code style used in most of
the Archivematica project.

sevein commented:
Thanks. Listing `1` feels redundant, could you try the following?

```json
{
    "exit_codes": {
        "0": {
            "job_status": "Completed successfully",
            "link_id": "b04e9232-2aea-49fc-9560-27349c8eba4e"
        }
    },
    "fallback_job_status": "Failed",
    "fallback_link_id": "61c316a6-0a50-4f65-8767-1f44b1eeb6dd"
}
```

This could be a simplification of the script:

```python
def call(jobs):
    for job in jobs:
        with job.JobContext():
            objects_dir = job.args[1]
            for _, _, files in scandir.walk(objects_dir):
                if files:
                    return
            job.set_status(1)
```

jorikvankemenade commented:
I agree that listing 1 is a bit redundant. I have removed it and it works without problems:

![image](https://user-images.githubusercontent.com/7934339/75111630-82c20000-563c-11ea-85ed-4ad61b64e7ab.png)

![image](https://user-images.githubusercontent.com/7934339/75111646-9a00ed80-563c-11ea-9441-5d618d7545d8.png)

![image](https://user-images.githubusercontent.com/7934339/75111653-aedd8100-563c-11ea-8946-2a1f3455e13a.png)

![image](https://user-images.githubusercontent.com/7934339/75111670-ee0bd200-563c-11ea-99ad-02d687f9affc.png)

I have also applied your simplification and added a docstring to the script.

sevein reviewed (approved):
Thanks!
